R/share_spec.R

```diff
@@ -129,49 +145,62 @@ share_spec.data.frame <- function(data,
                                                level_of_confidence_in_identification = "",
                                                other_info = "",
                                                license = "CC BY-NC"),
-                                  share = "system", ...) {
+                                  file = NULL,
+                                  share = "system",
+                                  id = paste(digest(Sys.info()),
+                                             digest(sessionInfo()), sep = "/"),
+                                  ...) {
   if (is.null(names(metadata))) stop("'metadata' needs to be a named vector")
   if (any(is.na(metadata[c("user_name", "spectrum_type", "spectrum_identity")])) |
       metadata["user_name"] == "" | metadata["spectrum_type"] == "" |
       metadata["spectrum_identity"] == "") {
     mex <- FALSE
     warning("fields 'user_name', 'spectrum_type', and 'spectrum_identity' ",
-            "must not be empty if you like to share your metadata")
+            "must not be empty if you like to share your metadata", call. = F)
   } else {
     mex <- TRUE
   }
 
-  id <- paste(human_ts(), digest(data, algo = "md5"), sep = "_")
-
+  fid <- digest(data)
   mdata <- data.frame(variable = names(metadata), input = metadata,
                       row.names = NULL)
 
   if (share == "system") {
-    pkg <- system.file("extdata", package = "OpenSpecy")
-    dir.create(file.path(pkg, "user_spectra"), showWarnings = F)
-    fp <- file.path(pkg, "user_spectra")
+    fp <- file.path(system.file("extdata", package = "OpenSpecy"),
```

Comment:
@zsteinmetz i think this will also need to be updated to tempdir() is that right?

Reply:
I wouldn't say so because we leave it to the user to chose that path ("system") or any other. As far as I understood, there is no general rule that every path should default to `tempdir()` (`write.csv()` doesn't do either), the user just needs to have a choice. 

What CRAN doesn't like is hardcoding any path and writing to it with internal functions of the package (like our automated tests) without the users knowledge.

So currently, `run_app()` is the only exception writing to and assuming library data in the package location. This might be even acceptable because it's a Shiny app and I've seen Shiny apps writing much more deliberately to the user space than other functions. Maybe CRAN would even notice because `run_app()` is not triggered by our automated tests also because it's hard to automatically test a Shiny app.